src/devices/colordistancesensor.ts

```diff
@@ -79,11 +81,128 @@ export class ColorDistanceSensor extends Device {
+     * @param {number} power -7 (full reverse) to 7 (full forward). 0 is stop. 8 is brake.
+     * @returns {Promise} Resolved upon successful issuance of the command.
+     */
+    public setPFPower (channel: number, output: string, power: number) {
+        let address = 0;
+        if (channel > 4) {
+            channel -= 4;
+            address = 1;
+        }
+        const message = Buffer.alloc(2);
+        // Send "Single output mode"
+        message[0] = ((channel - 1) << 4) + (address << 3) + (output === "RED" ? 4 : 5);
+        message[1] = this._pfPowerToPWM(power) << 4;
+        return this.sendPFIRMessage(message);
+    }
+
+
+    /**
+     * Start Power Functions motors running via IR
+     *
+     * NOTE: This command is designed for bang-bang style operation. To keep the motors running, the sensor needs to be within range of the IR receiver constantly.
+     * @method ColorDistanceSensor#startPFMotors
+     * @param {Buffer} channel Channel number, between 1-4
+     * @param {Buffer} powerA -7 (full reverse) to 7 (full forward). 0 is stop. 8 is brake.
+     * @param {Buffer} powerB -7 (full reverse) to 7 (full forward). 0 is stop. 8 is brake.
+     * @returns {Promise} Resolved upon successful issuance of the command.
+     */
+    public startPFMotors (channel: number, powerA: number, powerB: number) {
+        let address = 0;
+        if (channel > 4) {
+            channel -= 4;
+            address = 1;
+        }
+        const message = Buffer.alloc(2);
+        // Send "Combo PWD mode"
+        message[0] = (((channel - 1) + 4 + (address << 3)) << 4) + this._pfPowerToPWM(powerA);
+        message[1] += this._pfPowerToPWM(powerB) << 4;
+        return this.sendPFIRMessage(message);
+    }
+
+
+    /**
+     * Send a raw Power Functions IR command
+     * @method ColorDistanceSensor#sendPFIRMessage
+     * @param {Buffer} message 2 byte payload making up a Power Functions protocol command. NOTE: Only specify nibbles 1-3, nibble 4 should be zeroed.
+     * @returns {Promise} Resolved upon successful issuance of the command.
+     */
+    public sendPFIRMessage (message: Buffer) {
+        if (this.isWeDo2SmartHub) {
+            throw new Error("Power Functions IR is not available on the WeDo 2.0 Smart Hub");
+        } else {
+            const payload = Buffer.alloc(2);
+            payload[0] = (message[0] << 4) + (message[1] >> 4);
+            payload[1] = message[0] >> 4;
+            this.subscribe(Mode.PF_IR);
+            return this.writeDirect(0x07, payload);
+        }
+    }
+
+
+    /**
+     * Set the color of the LED on the sensor via a color value.
+     * @method ColorDistanceSensor#setColor
+     * @param {Color} color
+     * @returns {Promise} Resolved upon successful issuance of the command.
+     */
+    public setColor (color: number | boolean) {
+        return new Promise((resolve, reject) => {
+            if (typeof color === "boolean") {
```

Comment:
The intention is to allow `setColor(false)` - since color 0 turns off the LED, I felt it was intuitive to allow a boolean false to be passed in to turn off the LED.

Passing true would cause undefined behaviour though, so I think it'd be wise for me to add a guard around this.

Reply:
The intention could be more explicit with:
```
public setColor (color: number | false) {
  return new Promise((resolve, reject) => {
    if (color === false) {
      color = 0;
    }         
```
Or even (though I am a bit on the fence with the readability of it):
```
public setColor (color: number | false) {
  return new Promise((resolve, reject) => {
    color = color || 0;         
```